File: 14-longest-common-prefix/longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        strs.sort()
        
        if not strs:
            return ""

        if len(strs) == 1:
            return strs[0]

        result = ""

        first = strs[0]

        last = strs[-1]

        for index, char in enumerate(first):
            if (char != last[index]):
                return result
            result += char

        return result
        

File: 14-longest-common-prefix/longest-common-prefix.js
var longestCommonPrefix = function(strs) {
    strs = strs.sort()

    let first = strs[0]
    let last = strs.at(-1)

    if (!first) return ""

    if (first == last) return last

    for (let i = 0; i < last.length; i++) {
        if (first[i] != last[i]) return first.slice(0, i)
    }
    
    return first


};